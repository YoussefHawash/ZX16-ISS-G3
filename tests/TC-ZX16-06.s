.text
.org 0x0000
    j main

.org 0x0020
main:
    # Test LUI
    lui x1, 0x12         # x1 = 0x12 << 7 = 0x0900
    add x2, x1           # x2 = 0x0900 (just to verify)

    # Test AUIPC
    auipc x3, 0x01       # x3 = PC + (0x01 << 7)
                         # PC here is 0x0028 â†’ x3 = 0x0028 + 0x0080 = 0x00A8
    add x4, x3           # x4 = x3 (copied)

    ecall 0x00A