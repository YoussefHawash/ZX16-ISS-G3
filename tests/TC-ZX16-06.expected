J 32
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
LUI x1, 0x012
ADD x2, x1
AUIPC x3, 0x001
ADD x4, x3
ECALL 10    # ProgramExit
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0